service/src/main/resources/changelog/changesets/20230207_sarscov2_nextstrain_nonempty_optionals.yaml

```diff
@@ -0,0 +1,1312 @@
+                        "source": {
+                          "type": "none"
+                        }
+                      },
+                      {
+                        "input_name": "sarscov2_nextstrain.dedup_seqs.error_on_seq_diff",
+                        "input_type": {
+                          "type": "optional",
+                          "optional_type": {
+                            "type": "primitive",
+                            "primitive_type": "Boolean"
+                          }
+                        },
+                        "source": {
+                          "type": "none"
+                        }
+                      },
+                      {
+                        "input_name": "sarscov2_nextstrain.dedup_seqs.nextstrain_ncov_repo_commit",
+                        "input_type": {
+                          "type": "optional",
+                          "optional_type": {
+                            "type": "primitive",
+                            "primitive_type": "String"
+                          }
+                        },
+                        "source": {
+                          "type": "none"
+                        }
+                      },
+                      {
+                        "input_name": "sarscov2_nextstrain.derived_cols.docker",
+                        "input_type": {
+                          "type": "optional",
+                          "optional_type": {
+                            "type": "primitive",
+                            "primitive_type": "String"
+                          }
+                        },
+                        "source": {
+                          "type": "none"
+                        }
+                      },
+                      {
+                        "input_name": "sarscov2_nextstrain.derived_cols.lab_highlight_loc",
+                        "input_type": {
+                          "type": "optional",
+                          "optional_type": {
+                            "type": "primitive",
+                            "primitive_type": "String"
+                          }
+                        },
+                        "source": {
+                          "type": "none"
+                        }
+                      },
+                      {
+                        "input_name": "sarscov2_nextstrain.derived_cols.table_map",
+                        "input_type": {
+                          "type": "optional",
+                          "optional_type": {
+                            "type": "array",
+                            "array_type": {
+                              "type": "optional",
+                              "optional_type": {
+                                "type": "primitive",
+                                "primitive_type": "File"
+                              }
+                            },
+                            "non_empty": false
```

Comment:
since you can't see it in a diff: this is an example of the kind of change made in this changeset.

Previously, `non_empty` appeared as a property of `input_type` -- but `optional` doesn't have a `non_empty` property, so it caused an error.

this change moves `non_empty` to be a property of `optional_type`, which is an `array`